src/singletons/Toasts.hpp

```diff
@@ -35,6 +35,11 @@ class Toasts final
     void sendWindowsNotification(const QString &channelName,
                                  const QString &channelTitle, Platform p);
 
+    bool initialized_ = false;
+#elif defined(LIBNOTIFY)
+    void ensureInitialized();
+    void sendLibnotify(const QString &channelName, const QString &channelTitle);
+
     bool initialized_ = false;
```

Comment:
Could `ensureInitialized` and `initialized_`  be defined in a single block? They're the same on Windows and with libnotify. Maybe the `send*` functions could be united as well. Something like `sendPlatformNotification`? It would include the platform for now (https://github.com/Chatterino/chatterino2/pull/5914).

Reply:
I left these separate just to avoid getting into too much rework here, but combining them or moving them into a library seems good. The macos notification stuff required signing and other conditions that I didn't wade into. I don't boot windows up very often to test there either :laughing: 

For linux/freebsd, you could also avoid libnotify and use dbus directly, but a bit more involved to do.